extensions/amp-pinterest/0.1/follow-button.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Util} from './util';
+
+// Popup options
+const POP_FOLLOW = `status=no,resizable=yes,scrollbars=yes,
+  personalbar=no,directories=no,location=no,toolbar=no,
+  menubar=no,width=1040,height=640,left=0,top=0`;
+
+/*
+ * Pinterest Follow Button
+ * @attr data-href:  the url of the user's profile to follow
+ * @attr data-label: the text to display (user's full name)
+ **/
+export class FollowButton {
+
+  constructor(rootElement) {
+    AMP.assert(rootElement.getAttribute('data-href'),
+      'The data-href attribute is required for follow buttons');
+    AMP.assert(rootElement.getAttribute('data-label'),
+      'The data-label attribute is required for follow buttons');
+    this.element = rootElement;
+  }
+
+  /*
+   * Override the default href click handling to log and open popup
+   * @param {Event} evt: the HTML event object
+   **/
+  handleClick(evt) {
+    evt.preventDefault();
+    window.open(this.href, 'pin' + new Date().getTime(), POP_FOLLOW);
+    Util.log(`&type=button_follow&href=${this.href}`);
+  }
+
+  /*
+   * Convert a string into a DOM node
+   * @param {String} str: the string to convert
+   * @returns {Element}
+   **/
+  stringToDOM(str) {
+    const temp = document.createElement('span');
+    temp/*REVIEW*/.innerHTML = str;
```

Comment:
I'd say - let's perhaps make Util.make a better and safe version of itself, rather than doing string parsing. It's hard to reason about a possibility of XSS in the long-term since different pieces could be changing overtime. It's best not to introduce the risk in the first place. For instance, just few lines above there's code:
```
this.href = assertHttpsUrl(rootElement.getAttribute('data-href'));
```
I.e. the `href` comes from a user-entered value. I don't see it fully validated before it's injected, but even if it is, it's easy to loose that validation later.

I'm definitely sympathetic with the readability argument. But I think it can be achieved via something like a `make` method. Perhaps we can even then promote it to the core runtime for everyone else to use.